deepchem/models/torch_models/cgcnn.py

```diff
@@ -0,0 +1,255 @@
+      The length of the initial node feature vectors. The 92 is
+      based on length of vectors in the atom_init.json.
+    hidden_node_dim: int, default 64
+      The length of the hidden node feature vectors.
+    in_edge_dim: int, default 41
+      The length of the initial edge feature vectors. The 41 is
+      based on default setting of CGCNNFeaturizer.
+    num_conv: int, default 3
+      The number of convolutional layers.
+    predicator_hidden_feats: int, default 128
+      Size of hidden graph representations in the predicator, default to 128.
+    n_out: int
+      Number of the output size, default to 1.
+    """
+    super(CGCNN, self).__init__()
+    self.embedding = nn.Linear(in_node_dim, hidden_node_dim)
+    self.convs = [
+        CGCNNLayer(
+            hidden_node_dim=hidden_node_dim,
+            edge_dim=in_edge_dim,
+            batch_norm=True) for _ in range(num_conv)
+    ]
+    self.fc = nn.Linear(hidden_node_dim, predicator_hidden_feats)
+    self.out = nn.Linear(predicator_hidden_feats, n_out)
+
+  def forward(self, dgl_graph):
+    """Predict labels
+
+    Parameters
+    ----------
+    dgl_graph: DGLGraph
+      DGLGraph for a batch of graphs. The graph expects that the node features
+      are stored in `ndata['x']`, and the edge features are stored in `edata['edge_attr']`.
+
+    Returns
+    -------
+    out: torch.Tensor
+      The output value, the shape is `(batch_size, n_out)`.
+    """
+    try:
+      import dgl
+    except:
+      raise ValueError("This class requires DGL to be installed.")
+
+    graph = dgl_graph
+    # embedding node features
+    graph.ndata['x'] = self.embedding(graph.ndata['x'])
+
+    # convolutional layer
+    for conv in self.convs:
+      graph = conv(graph)
+
+    # pooling
+    graph_feat = dgl.mean_nodes(graph, 'x')
+    graph_feat = self.fc(graph_feat)
+    out = self.out(graph_feat)
+    return out
+
+
+class CGCNNModel(TorchModel):
+  """CGCNN wrapper model for converting PyTorch style to Keras style.
+
+  Please confirm the details about CGCNN from CGCNN class docstring.
+  Here is a simple example of code that uses the CGCNNModel with
+  materials dataset.
+
+  >> import deepchem as dc
+  >> dataset_config = {"reload": False, "featurizer": dc.feat.CGCNNFeaturizer, "transformers": []}
+  >> tasks, datasets, transformers = dc.molnet.load_perovskite(reload=False)
+  >> train, valid, test = datasets
+  >> model = dc.models.CGCNNModel(loss=dc.models.losses.L2Loss(), batch_size=32, learning_rate=0.001)
+  >> model.fit(train, nb_epoch=50)
+
+  Notes
+  -----
+  This class requires DGL and PyTorch to be installed.
+  """
+
+  def __init__(self,
+               in_node_dim: int = 92,
+               hidden_node_dim: int = 64,
+               in_edge_dim: int = 41,
+               num_conv: int = 3,
+               predicator_hidden_feats: int = 128,
+               n_out: int = 1,
+               **kwargs):
+    """
+    Parameters
+    ----------
```

Comment:
We should add a note here that it also accepts all the arguments TorchModel does.

Reply:
I added!

Reply:
You don't need to list all the arguments individually.  That list will quickly get out of date as we add or change arguments to TorchModel.  Instead you can just add a single line at the top of the docstring:

> In addition to the following arguments, this class also accepts all the keyword arguments from TorchModel.

Reply:
Oh, sorry. I misunderstood your comment. I fixed the docstring.